src/pygef/cpt.py

```diff
@@ -2,13 +2,26 @@
 
 import copy
 import pprint
-from dataclasses import dataclass, field
+from dataclasses import field
 from datetime import date
 from enum import Enum
 from typing import Any, List
 
 import polars as pl
 
+# check if pydantic is installed
+try:
```

Comment:
@tversteeg I tried to keep pydantic as an optional dependency because I wasn't sure if you were willing to accept pydantic as an additional dependency. 

Pydantic offers a near drop-in replacement for the builtins `dataclass` decorator, so thought this might be an elegant solution. Upon running the tests there are so validation errors where some fields were `None` while this was not allowed according to the type annotations,

Still need a some work of course to convert the `polars` DataFrame to json, but wanted to get a quick check on how you feel about using pydantic in this manner.

Reply:
I like this solution, it looks elegent. Although I wonder if it needs to be optional, is there a convincing argument against using pydantic?

Reply:
Not for me, I use Pydantic all over the place so would recommend it :)
But I think it will introduce a small overhead due to data validation step introduced by Pydantic.

Reply:
But that overhead also prevents some nasty bugs and user errors, so for me that's probably worth it. @RDWimmers @tlukkezen, thoughts?

Reply:
yes, lets add pydantic as an additional dependency.